modules/core/src/smithy4s/Transformation.scala

```diff
@@ -120,10 +120,12 @@ object Transformation {
     }
 
   implicit def mappedHintsServiceTransform[Alg[_[_, _, _, _, _]]]: Transformation[Hints=>Hints, Service[Alg], Service[Alg]] =
-    (func: Hints => Hints, that: Service[Alg]) => new Service[Alg] {
+    (mapper: Hints=>Hints, that: Service[Alg]) => new Service[Alg] {
       override type Operation[I, E, O, SI, SO] = that.Operation[I, E, O, SI, SO]
 
-      private val cache = that.endpoints.map(e => e.id -> e.mapHints(func)).toMap
+      private val cache = that.endpoints.foldLeft(Map.empty[ShapeId, Endpoint[I, E, O, SI, SO]]) {
+        case (map, endpoint) => map.+(endpoint.id -> endpoint.mapHints(mapper))
+      }
```

Comment:
nitpick: this may be pretty inefficient, creating N maps for N elements. Might want to use a builder instead, although I doubt this will be in the hot path anyway

Reply:
agreed , I didn't see this occurring on the hot path , however will switch a builder regardless